### PR TITLE
[RA2 Ch2,Ch4]: Kubernetes API feature requirements

### LINKED ISSUE #2349: [RA2 Ch4] Define usage policy for Kubernetes alpha and beta API-s
Kubernetes have lots of non GA API-s which are not production ready and can be deprecated at any time. I think we should define an explicit policy about the usage of alfa and beta api-s. 

I would propose that: 
 - beta API-s are okey to use
 - An explicit selected list of alfa API-s are okey to use or just restrict the usage of alfa API-s

Any opinions?

### PR DESCRIPTION
**Pull request by CsatariGergely**
Adding requirements to support all GA and beta Kubernetes features, but
to disable all Alpha features.

Closes #2349

### CODE REVIEW

**Review by tomkivlin — COMMENTED** [review]
Is it worth clarifying the relationship between feature gates and APIs?  e.g. to ensure we don't specify an Alpha feature gate in the spec, which in turn enables an Alpha API?

Or, do we need to also specify no Alpha feature gates, to ensure that those Feature Gates that don't relate to a new API (e.g. the [SuspendJob API](https://kubernetes.io/docs/concepts/workloads/controllers/job/#suspending-a-job) which enables a feature within the Batch API.

**Comment by CsatariGergely** [comment]
> 
> 
> Is it worth clarifying the relationship between feature gates and APIs? e.g. to ensure we don't specify an Alpha feature gate in the spec, which in turn enables an Alpha API?
> 
> Or, do we need to also specify no Alpha feature gates, to ensure that those Feature Gates that don't relate to a new API (e.g. the [SuspendJob Feature Gate](https://kubernetes.io/docs/concepts/workloads/controllers/job/#suspending-a-job) which is Alpha, but enables a feature within the Batch API, which is not Alpha.

Hopefully clarified in bb11f9e

File: doc/ref_arch/kubernetes/chapters/chapter02.md [hunk 1]
```diff
@@ -310,6 +310,8 @@ The requirements in this section are to be delivered in addition to those in [se
 | `req.kcm.gen.02` | Kubernetes Cluster | General | The Architecture **must** enable workload resiliency. |[ra2.k8s.004](chapter04.md#43-kubernetes)|
 | `req.int.api.01` | API | General | The Architecture **must** leverage the Kubernetes APIs to discover and declaratively manage compute (virtual and bare metal resources), network, and storage. |For Networking: <ul><li>[ra2.ntw.001](chapter04.md#45-networking-solutions)<li>[ra2.ntw.008](chapter04.md#45-networking-solutions)<li>[ra2.app.006](chapter04.md#49-kubernetes-workloads)</ul><br>Compute/storage not yet met. |
 | `req.int.api.02` | API | General | The Architecture **must** support the usage of a Kubernetes Application package manager using the Kubernetes API, like Helm v3. |[ra2.pkg.001](chapter04.md#48-kubernetes-application-package-manager)|
+| `req.int.api.03` | API | General | The Architecture **must** support stable features in its API-s. ||
```
Comment (rgstori) [inline doc/ref_arch/kubernetes/chapters/chapter02.md:313]:
```suggestion
| `req.int.api.03` | API | General | The Architecture **must** support stable features in its APIs. ||
```
typo: API-s to APIs

File: doc/ref_arch/kubernetes/chapters/chapter02.md [hunk 2]
```diff
@@ -310,6 +310,8 @@ The requirements in this section are to be delivered in addition to those in [se
...
+| `req.int.api.03` | API | General | The Architecture **must** support limited backward compatibility in its API-s. Support for the whole API must not be dropped, but the schema or other details can change. ||
```
Comment (rgstori) [inline doc/ref_arch/kubernetes/chapters/chapter02.md:314]:
```suggestion
| `req.int.api.03` | API | General | The Architecture **must** support limited backward compatibility in its APIs. Support for the whole API must not be dropped, but the schema or other details can change. ||
```
typo

File: doc/ref_arch/kubernetes/chapters/chapter04.md [hunk 3]
```diff
@@ -84,6 +84,8 @@ In order for the Kubernetes components to be conformant with the Reference Archi
...
 |`ra2.k8s.011`|Anuket profile labels|To clearly identify which worker nodes are compliant with the different profiles defined by Anuket the worker nodes must be labelled according to the following pattern: an `anuket.io/profile/basic` label must be set to `true` on the worker node if it can fulfil the requirements of the basic profile and an `anuket.io/profile/network-intensive` label must be set to `true` on the worker node if it can fulfil the requirements of the network intensive profile. The requirements for both profiles can be found in [chapter 2](./chapter02.md#22-reference-model-requirements)|||
+|`ra2.k8s.012`|Kubernetes API-s|Kubernetes [Alpha API](https://kubernetes.io/docs/reference/using-api/#api-versioning) are recommended only for testing, therefore all Alpha API-s **must** be disabled.|[req.int.api.03](./chapter02.md#22-reference-model-requirements)||
```
Comment (rgstori) [inline doc/ref_arch/kubernetes/chapters/chapter04.md:87]:
```suggestion
|`ra2.k8s.012`|Kubernetes APIs|Kubernetes [Alpha API](https://kubernetes.io/docs/reference/using-api/#api-versioning) are recommended only for testing, therefore all Alpha APIs **must** be disabled.|[req.int.api.03](./chapter02.md#22-reference-model-requirements)||
```

File: doc/ref_arch/kubernetes/chapters/chapter04.md [hunk 4]
```diff
@@ -84,6 +84,8 @@ In order for the Kubernetes components to be conformant with the Reference Archi
...
+|`ra2.k8s.013`|Kubernetes API-s|To provide backward compatibility all supported GA and Beta API-s of Kubernetes **must** be enabled. |[req.int.api.04](./chapter02.md#22-reference-model-requirements)||
```
Comment (rgstori) [inline doc/ref_arch/kubernetes/chapters/chapter04.md:88]:
```suggestion
|`ra2.k8s.013`|Kubernetes APIs|Backward compatibility of all supported GA and Beta APIs of Kubernetes **must** be supported. |[req.int.api.04](./chapter02.md#22-reference-model-requirements)||
```
clarified

**Review by rgstori — CHANGES_REQUESTED** [review]

**Review by rgstori — APPROVED** [review]

**Review by petorre — APPROVED** [review]

**Review by walterkozlowski — APPROVED** [review]

**Pull request merged**